src/services/assetService.ts

```diff
@@ -144,11 +144,19 @@ export class AssetService {
         }
     }
 
-    // If extension of a file was spoofed, we fetch only first 4 bytes of the file and read MIME type
+    // If extension of a file was spoofed, we fetch only first 4 or needed amount of bytes of the file and read MIME type
     public static async getMimeType(uri: string): Promise<string[]> {
-        const first4bytes: Response = await fetch(uri, { headers: { range: `bytes=0-${mimeBytesNeeded}` } });
+        const getFirst4bytes = (): Promise<Response> => this.pollForFetchAPI(() => fetch(uri, {headers: {range: `bytes=0-${mimeBytesNeeded}`}}), 1000, 200);
+        let first4bytes: Response;
+        try {
+            first4bytes = await getFirst4bytes()
+        } catch {
+            return new Promise<string[]>((resolve) => {
+                resolve([]);
```

Comment:
We shouldn't resolve with empty array as that represents invalid mime type for a failed fetch and will show an incorrect invalid mime type error